kong/init.lua

```diff
@@ -884,12 +884,18 @@ function Kong.init_worker()
     kong.cache:invalidate_local(constants.ADMIN_GUI_KCONFIG_CACHE_KEY)
   end
 
-  if process.type() == "privileged agent" and not kong.sync then
-    if kong.clustering then
-      -- full sync cp/dp
+  if kong.clustering then
+    -- full sync dp
+    -- "privileged agent" must be a DP worker
+    if process.type() == "privileged agent" and not kong.sync then
```

Comment:
I think that it will be easy to understand:
```
local is_cp = is_control_plane(kong.configuration)
local is_privileged_dp = process.type() == "privileged agent" and not kong.sync
if is_cp or is_privileged_dp then
	kong.clustering:init_worker()
end
if is_privileged_dp then
	return
end
```